### PR TITLE
enhance: reduce RPC calls while loading

### LINKED ISSUE #29497: [Enhancement]: Remove the RPC `DescribeIndex` for each load segment request
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

We could fetch the index info while fetching the new target, then we could remove the RPC on load segment path

### Why is this needed?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
the index info could be saved as Milvus doesn't support to alter it after loaded, we don't need to get it for every segment/channel

fix #29497

### CODE REVIEW

File: internal/querycoordv2/meta/target_manager.go [hunk 1]
```diff
@@ -214,17 +213,22 @@ func (mgr *TargetManager) PullNextTargetV2(broker Broker, collectionID int64, ch
...
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	return segments, dmChannels, nil
+	indexes, err := broker.DescribeIndex(context.Background(), collectionID)
+	if err != nil {
+		return nil, err
+	}
```
Comment (congqixia) [inline internal/querycoordv2/meta/target_manager.go:229]: shall we put it into retry scope?

Comment (yah01) [reply internal/querycoordv2/meta/target_manager.go:229]: no need to retry manually, the observer would try this constantly

**Pull request closed without merging**